### ISSUE TITLE
Theoretical problem: what's the meaning of the symbol M and how to learn it in your second architecture RCGAN-U? 

### ISSUE THREAD

**Issue by rosefun** [POLane16/Robust-Conditional-GAN#2]
I'm still confused about your second architecture RCGAN-U. As we can not get the confuse matrix C, do you use matrix M as the estimation of matrix C?

![image](https://user-images.githubusercontent.com/32197447/67063865-e32ac780-f19a-11e9-874d-ae04d379c63e.png)

![image](https://user-images.githubusercontent.com/32197447/67063880-f0e04d00-f19a-11e9-9145-3b2f3b87c389.png)

Besides, does matrix M be learned by the model ?

Hopefully for reply!

**Comment by tkkiran** [comment]
Yes, you are right. M is the estimated confusion matrix for C, whose rows sum to one, sum_{j=1, 2, ..., m} M_{ij} = 1  (each row should be a probability vector). We enforce this property by using a softmax laye.
https://github.com/POLane16/Robust-Conditional-GAN/blob/63e60084667fd426a9cf7a43283dd0ddc4d80541/mnist/model.py#L103

Yes, RCGAN-U simultaneously learns M and G by back propagating from the above loss function to matrix M and the parameters of the generator neural net G. Notice that from the second expression the loss function function is linear in M.

Although we don't have any theoretical guarantees for learning M, empirically we observed that RCGAN-U learns a good model G and a confusion matrix M close enough to C.

Hope this helps. I am closing this, but feel free to ask more question here.
